src/app/components/app-terminal/app-terminal.component.ts

```diff
@@ -18,183 +21,57 @@ import { SlimScrollEvent, ISlimScrollOptions } from 'ngx-slimscroll';
 export class AppTerminalComponent implements OnInit {
   @Input() data: any;
   @Input() options: { size: 'normal' | 'large' };
-  au: any;
-  commands: { command: string, visible: boolean, output: string, time: string }[];
-  noData: boolean;
-  initScroll: boolean;
-  scrollOptions: ISlimScrollOptions;
-  scrollEvents: EventEmitter<SlimScrollEvent>;
+  hterm: hterm.Terminal;
 
   constructor(
     private elementRef: ElementRef,
     @Inject(DOCUMENT) private document: any
   ) {
-    this.scrollOptions = {
-      barBackground: '#666',
-      gridBackground: '#000',
-      barBorderRadius: '10',
-      barWidth: '7',
-      gridWidth: '7',
-      barMargin: '2px 5px',
-      gridMargin: '2px 5px',
-      gridBorderRadius: '10',
-      alwaysVisible: false
-    };
-
-    this.scrollEvents = new EventEmitter<SlimScrollEvent>();
+    hterm.hterm.defaultStorage = new hterm.lib.Storage.Local();
+    this.hterm = new hterm.hterm.Terminal();
   }
 
   ngOnInit() {
-    this.au = new AnsiUp.default();
-    this.au.use_classes = true;
-    this.commands = [];
-    this.noData = true;
+    this.hterm.onTerminalReady = () => {
+      this.hterm.setWindowTitle = () => {};
+      this.hterm.prefs_.set('cursor-color', 'transparent');
+      this.hterm.prefs_.set('font-family', 'monaco, menlo, monospace');
+      this.hterm.prefs_.set('font-size', 11);
+      this.hterm.prefs_.set('audible-bell-sound', '');
+      this.hterm.prefs_.set('font-smoothing', 'subpixel-antialiased');
+      this.hterm.prefs_.set('enable-bold', false);
+      this.hterm.prefs_.set('backspace-sends-backspace', true);
+      this.hterm.prefs_.set('cursor-blink', false);
+      this.hterm.prefs_.set('receive-encoding', 'raw');
+      this.hterm.prefs_.set('send-encoding', 'raw');
+      this.hterm.prefs_.set('alt-sends-what', 'browser-key');
+      this.hterm.prefs_.set('scrollbar-visible', false);
+      this.hterm.prefs_.set('enable-clipboard-notice', false);
+      this.hterm.prefs_.set('background-color', '#000000');
+      this.hterm.prefs_.set('foreground-color', '#f8f8f2');
+      this.hterm.prefs_.set('color-palette-overrides', terminalColorPallete);
+      this.hterm.onVTKeystroke = () => {};
+    };
+
+    this.hterm.decorate(document.querySelector('#hterm'));
```

Comment:
use `this.document` that we have injected instead of global browser variable `document`.

Reply:
also, why is `#hterm` needed to be referenced by adding additional `id` property if you can call it with class name?